platform-plugins.md

```diff
@@ -0,0 +1,277 @@
+---
+layout: page
+title: Platform-specific APIs and Platform Plugins
```

Comment:
Would we add an example of binary data to this doc? At a minimum, being more clear that it's possible and when I'd use it.

Do we know enough to document patterns of use? Are there common themes/patterns for interop for various different APIs? Some APIs are stateful, some are stateless (like battery). How does that impact my interop design?